internal/services/api/post_packages_r_scan.go

```diff
@@ -52,24 +57,18 @@ func (h *PostPackagesRScanHandler) ServeHTTP(w http.ResponseWriter, req *http.Re
 	}
 	// Can't call ValidateFilename on b.SaveName because
 	// it may contain slashes.
-	path := util.NewRelativePath(filepath.FromSlash(b.SaveName), nil)
-	err = util.ValidateFilename(path.Base())
+	lockfileRelPath := util.NewRelativePath(filepath.FromSlash(b.SaveName), nil)
+	err = util.ValidateFilename(lockfileRelPath.Base())
 	if err != nil {
 		BadRequest(w, req, h.log, err)
 		return
 	}
-	lockfileAbsPath := projectDir.Join(path.String())
-	rPath := util.NewPath(b.R, nil)
-
-	rInterpreter, err := h.rInterpreterFactory(projectDir, rPath, h.log, nil, nil, nil)
-	if err != nil {
-		InternalError(w, req, h.log, err)
-		return
-	}
-	err = rInterpreter.CreateLockfile(lockfileAbsPath)
+	dependencyScanner := h.rDependencyScannerFactory(h.log)
+	_, err = dependencyScanner.SetupRenvInDir(projectDir.String(), lockfileRelPath.String(), b.R)
```

Comment:
This seems to be necessary only for tests, but adding complexity to the implementation to make tests easier does seem something that will hit back on long term.

The `PostPackagesRScanHandler` could have the scanner as an attribute instead of the factory, the scanner is primarily stateless, it only relies on the Executor and the Executor has no state. So we can just pass a scanner to `NewPostPackagesRScanHandler` and avoid the factory.

To be honest, I think I'd make the scanner an internal detail of `PostPackagesRScanHandler` and have the `NewPostPackagesRScanHandler` create one by itself. At that point the tests could just change the attribute of the resulting post handler instead of passing arguments to the constructor.